lacework/resource_lacework_integration_ecr.go

```diff
@@ -1,13 +1,51 @@
 package lacework
 
 import (
+	"fmt"
 	"log"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
 
 	"github.com/lacework/go-sdk/api"
 )
 
+func importLaceworkECRIntegration(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	lacework := meta.(*api.Client)
+
+	log.Printf("[INFO] Importing Lacework integration with guid: %s\n", d.Id())
+	response, err := lacework.Integrations.Get(d.Id())
+	if err != nil {
+		return nil, err
+	}
+
+	for _, integration := range response.Data {
+		if integration.IntgGuid == d.Id() {
+			log.Printf("[INFO] Integration found with guid: %v\n", integration.IntgGuid)
+
+			// @afiune this field is important since it tells the resource which API functions to use
+			// we will extract it from the raw integration response so that the READ functions populate
+			// the rest of the fiels
+			if awsAuthType, ok := integration.Data["AWS_AUTH_TYPE"]; ok {
```

Comment:
We don't use the regular integration importer func since for ECR integrations we must know what the Authentication method is to use the appropriate GO-SDK functions.